armory/cli/tools/collect_outputs.py

```diff
@@ -0,0 +1,581 @@
+        and d.stat().st_mtime - time.time() > expiration
+    ]
+    log.info(
+        f"Found {len(empty_dirs)} empty directories to clean. Moving to {clean_dir}"
+    )
+    for d in empty_dirs:
+        target = clean_dir / d.relative_to(output_dir)
+        target.parent.mkdir(parents=True, exist_ok=True)
+        d.rename(target)
+    # delete all pairs of .png files NAME.png and NAME_adv.png where both files are equivalent
+    # (i.e. NAME.png is a copy of NAME_adv.png)
+    log.info("Cleaning up duplicate poisoned .png files")
+
+    pngs = list(
+        filter(
+            lambda p: len(list(p.parent.parent.glob("*Poison*"))) > 0,
+            output_dir.rglob("*x.png"),
+        )
+    )
+    log.debug(f"Found {len(pngs)} .png files to check for duplicates.")
+    unlinks = set()
+    for png in simple_progress_bar(pngs, msg="Checking equality..."):
+        adv = png.parent / (png.stem + "_adv.png")
+        if adv.exists() and Image.open(png).tobytes() == Image.open(adv).tobytes():
+            unlinks.add(png)
+            unlinks.add(adv)
+    log.info(
+        f"Found {len(pngs) - len(unlinks) // 2}/{len(pngs)} non-duplicate .png pairs. Moving the rest to {clean_dir}/pngs/"
+    )
+    for png in simple_progress_bar(unlinks, msg="Moving files..."):
+        target = clean_dir / png.relative_to(output_dir)
+        target.parent.mkdir(parents=True, exist_ok=True)
+        png.rename(target)
+    log.info(
+        f"Moved {len(unlinks) // 2} pairs of duplicate .png files to {clean_dir}/pngs/"
+    )
+
+
+def _get_run_name(d) -> str:
+    fallthrough = [
+        "config.sysconfig.filepath",
+        "config.sysconfig.config_filepath",
+        "config.sysconfig.output_filename",
+    ]
+    filepath = None
+    while filepath is None and len(fallthrough) > 0:
+        filepath = _get_dict_value(fallthrough.pop(0), d, default_value=None)
+    if filepath is None:
+        # can't find config filepath, just return result filepath as %s
+        return "%s"
+    p_filepath = Path(filepath)
+    if p_filepath.is_absolute():
+        common = os.path.commonpath([Path.cwd(), p_filepath])
+        filepath = p_filepath.relative_to(common)
+    return f"[{filepath}]({filepath})"
+    # return f"[{filepath}](%s)"
+
+
+def _get_dataset_name(d) -> str:
+    name = d["config"]["dataset"].get("name", None)
+    if name is None:
+        name = d["config"]["dataset"].get("test", {}).get("name", None)
+    return name
+
+
+def _get_attack_kwargs(d) -> str:
+    name_map = {
+        "learning_rate": "lr",
+        "learning_rate_depth": "lrd",
+        # "max_iter": "max_iter",
+        "patch_base_image": "base_image",
+        "learning_rate_schedule": "lrs",
+    }
+    ignore_keys = {
+        "optimizer",
+        "targeted",
+        "verbose",
+        "patch_base_image",
+        "patch",
+        "device_name",
+        "batch_size",
+    }
+    _ = d["config"]["attack"]["kwargs"].pop("patch_mask", None)  # remove patch_mask
```

Comment:
Can `patch_mask` be added to `ignore_keys` instead of popping?

Reply:
The kwargs section needs cleaned up; `patch_mask` should be included in the output. Will address shortly

Reply:
Addressed by https://github.com/twosixlabs/armory/pull/1986/commits/723c810c601b72f36d4db2bc5894b8f53692dcc0